javatests/arcs/android/integration/policy/PolicyTest.kt

```diff
@@ -80,27 +82,90 @@ class PolicyTest {
     assertOnlySingletonAbIsPersisted(startingKey)
   }
 
-  private suspend fun assertOnlySingletonAbIsPersisted(startingKey: StorageKey) {
+  /**
+   * Scenario: A policy-compliant recipe with @ttl handles egresses data and stores
+   * ingress-restricted values to RAM that are deleted after the TTL expires and at the runtime’s
+   * end.
+   */
+  @Test
+  fun ttlHandlesWithEgress_StoresIngressRestrictedValues_deletedAtTtlTimeAndRuntimeEnd() =
+    runBlocking {
+      env.addNewHostWith(
+        ::IngressThing.toRegistration(),
+        ::EgressAB.toRegistration()
+      )
+
+      // When the Arc is run...
+      val arc = env.startArc(TtlEgressesPlan)
+      env.waitForIdle(arc)
+
+      val ingest = env.getParticle<IngressThing>(arc)
+      val egressAB = env.getParticle<EgressAB>(arc)
+
+      withTimeout(30000) {
```

Comment:
I might have asked this earlier, but I can't remember what the answer was. Is using a timeout OK? Isn't there a better way to wait for completion?

Reply:
Yeah, I think we opted to keep the timeout in the last discussion. This is just a safeguard, the ingestion should happen right away (unless something is going seriously wrong).